src/Microsoft.OData.Core/UriParser/Parsers/SegmentArgumentParser.cs

```diff
@@ -275,7 +283,7 @@ private static bool TryParseFromUri(string text, out SegmentArgumentParser insta
             List<string> positionalValues = null;
 
             // parse keys just like function parameters
-            ExpressionLexer lexer = new ExpressionLexer(string.Concat("(", text, ")"), true, false);
+            ExpressionLexer lexer = new ExpressionLexer(string.Concat("(", text, ")"), true, false, true);

```

Comment:
@mikepizzo We should use named parameters here - especially for the boolean arguments - to improve readability and make the intent of each parameter clearer:
```
ExpressionLexer lexer = new ExpressionLexer(
    string.Concat("(", text, ")"),
    moveToFirstToken: true,
    useSemicolonDelimiter: false,
    parsingFunctionParameters: true);
```